### DISCUSSION TITLE
LSP: add a way to "list all occurences" of a struct or enum or class or object in a picker

### DISCUSSION THREAD

**Discussion by jerabaul29** [helix-editor/helix#9238] [Q&A] [answered]
Is it correct that there is no way at present to do a LSP server request to list all occurences of an object in the current project? For example asking rust-analyzer to "list / open a picker of all occurences of a struct"?

If this is correct, is it something that could be added?

---

For now I do things like:

```space``` + ```/``` + TO_SEARCH

which works well enough, but is not aware of types etc. Having some form of similar picker that is querying the LSP server would be awesome :) .

**Comment by gabydd** [comment, marked as answer]
Does `gr` goto references work for you?

**Reply by jerabaul29** [reply]
Yes, this is what I was looking for :) . Many thanks :) .
